tests/unit/__init__.py

```diff
@@ -0,0 +1,89 @@
+#tests/unit/__init__.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""NVDA unit testing.
+All unit tests should reside within this package and should be
+divided into modules and packages similar to the code they are testing.
+Test modules must have a C{test_} prefix
+and should contain one or more classes with a C{Test} prefix which subclass L{TestCase}.
+Methods in test classes should have a C{test_} prefix.
+"""
+
+### Ugly bootstrapping code.
+
+import os
+import sys
+
+# The path to the unit tests.
+UNIT_DIR = os.path.dirname(os.path.abspath(__file__))
+# The path to the top of the repo.
+TOP_DIR = os.path.dirname(os.path.dirname(UNIT_DIR))
+# The path to the NVDA "source" directory.
+SOURCE_DIR = os.path.join(TOP_DIR, "source")
+# Let us import modules from the NVDA source.
+sys.path.insert(1, SOURCE_DIR)
+import sourceEnv
+
+# Set options normally taken from the command line.
+import globalVars
+class AppArgs:
+	# The path from which to load a configuration file.
+	# Ideally, this would be an in-memory, default configuration.
+	# However, config currently requires a path.
+	# We use the unit test directory, since we want a clean config.
+	configPath = UNIT_DIR.decode("mbcs")
+	secure = False
+	disableAddons = True
+globalVars.appArgs = AppArgs()
+
+# We depend on the current directory to load some files;
+# e.g. braille imports louis which loads liblouis.dll using a relative path.
+os.chdir(SOURCE_DIR)
+# The path to this package might be relative, so make it absolute,
+# since we just changed directory.
+__path__[0] = UNIT_DIR
+# We don't want logging.
+import logging
+from logHandler import log
+log.addHandler(logging.NullHandler())

```

Comment:
As discussed, I think it could be helpful to be able to get the log information out of some tests. The log info could make debugging the unit tests easier. Could you open (and p4) an issue to keep track of the different ways we talked about to achieve this? I think for now there is low urgency on actually implementing this, however at the point when someone is trying to debug a unit test this could become more important.